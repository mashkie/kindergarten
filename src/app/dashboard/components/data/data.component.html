<div class="data">
  <table class="data__table">
    <tr class="data__table__row">
      <th class="data__table__row__header">Child Name</th>
      <th class="data__table__row__header">Kindergarden Name</th>
      <th class="data__table__row__header">Kindergarden Address</th>
      <th class="data__table__row__header">Child Age</th>
      <th class="data__table__row__header">Birthday of Child</th>
    </tr>
    <tr class="data__table__row" *ngFor="let child of children">
      <td class="data__table__row__data">{{child.name}}</td>
      <td class="data__table__row__data">{{(child.kindergartenId | kindergarden: kindergardens!)?.name}}</td>
      <td class="data__table__row__data">{{(child.kindergartenId | kindergarden: kindergardens!)?.address}}</td>
      <td class="data__table__row__data">{{child.birthDate | age}}</td>
      <td class="data__table__row__data">{{child.birthDate | date}}</td>
    </tr>
  </table>

</div>
