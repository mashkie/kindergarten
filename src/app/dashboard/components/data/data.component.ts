import { Component, Input } from '@angular/core';
import { Child } from '../../models/child.model';
import { Kindergarden } from '../../models/kindergarden.model';

@Component({
  selector: 'app-data',
  templateUrl: './data.component.html',
  styleUrls: ['./data.component.scss'],
})
export class DataComponent {
  @Input() children?: Child[];
  @Input() kindergardens?: Kindergarden[];

  getKindergardenFromChild(id: string): Kindergarden | undefined {
    return this.kindergardens?.find((kindergarden) => kindergarden.id === id);
  }
}
