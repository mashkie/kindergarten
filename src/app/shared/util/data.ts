export const kindergardens = [
  {
    id: 0,
    name: 'Waldorfkindergarten im Zentrum Wiens',
    address: 'Oleandergasse 21, 1220 Wien',
    betreiber: 'MA 10 - Wiener Kindergärten',
    typ: 'privat',
  },
  {
    id: 1,
    name: 'Kindergarten der Karl Schubert Schule für seelenpflegebedürftige Kinder und Jugendliche in Wien',
    address: 'Kanitzgasse 1-3, 1230 Wien',
    betreiber:
      'Verein Karl Schubertschule für Seelenpflege- bedürftige Kinder und Jugendliche in Wien',
    typ: 'privat',
  },
  {
    id: 2,
    name: 'Kindergarten der Stadt Wien',
    address: 'Audorfgasse 20, 1210 Wien',
    betreiber: 'MA 10 - Wiener Kindergärten',
    typ: 'öffentlich',
  },
  {
    id: 3,
    name: 'Kindergarten der Stadt Wien',
    address: 'Moßbachergasse 20-24, 1140 Wien',
    betreiber: 'MA 10 - Wiener Kindergärten',
    typ: 'öffentlich',
  },
  {
    id: 4,
    name: 'Montessori Kinderhaus Wien 1',
    address: 'Nußdorferstraße 6, 1090 Wien',
    betreiber: 'Montessori Kinderhaus Wien',
    typ: 'privat',
  },
];

export var children = [
  {
    id: '9b1deb4d-3b7d-4bad-9bpp-2b0d7b3dcb6d',
    name: 'Nicole Fitsch',
    birthDate: '2021-07-04',
    kindergarten: {
      id: 0,
      name: 'Waldorfkindergarten im Zentrum Wiens',
      address: 'Oleandergasse 21, 1220 Wien',
      betreiber: 'MA 10 - Wiener Kindergärten',
      typ: 'privat',
    },
  },
  {
    id: '9b1deb4d-897d-4bad-9bdd-2b0d7b3dcb6d',
    name: 'Jasmin Schmied',
    birthDate: '2020-09-04',
    kindergarten: {
      id: 0,
      name: 'Waldorfkindergarten im Zentrum Wiens',
      address: 'Oleandergasse 21, 1220 Wien',
      betreiber: 'MA 10 - Wiener Kindergärten',
      typ: 'privat',
    },
  },
  {
    id: '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d',
    name: 'Melanie Hollunder',
    birthDate: '2020-07-04',
    kindergarten: {
      id: 1,
      name: 'Kindergarten der Karl Schubert Schule für seelenpflegebedürftige Kinder und Jugendliche in Wien',
      address: 'Kanitzgasse 1-3, 1230 Wien',
      betreiber:
        'Verein Karl Schubertschule für Seelenpflege- bedürftige Kinder und Jugendliche in Wien',
      typ: 'privat',
    },
  },
  {
    id: '1b9d6bcd-bbfd-4b2d-9b5d-ab8dfbbd4bed',
    name: 'VIktor Sachs',
    birthDate: '2022-06-04',
    kindergarten: {
      id: 2,
      name: 'Kindergarten der Stadt Wien',
      address: 'Audorfgasse 20, 1210 Wien',
      betreiber: 'MA 10 - Wiener Kindergärten',
      typ: 'öffentlich',
    },
  },
  {
    id: '9b1debop-3b7d-4bad-9bdd-2b0d7b3dcb6d',
    name: 'Emily Nilson',
    birthDate: '2019-01-04',
    kindergarten: {
      id: 3,
      name: 'Kindergarten der Stadt Wien',
      address: 'Moßbachergasse 20-24, 1140 Wien',
      betreiber: 'MA 10 - Wiener Kindergärten',
      typ: 'öffentlich',
    },
  },
];
