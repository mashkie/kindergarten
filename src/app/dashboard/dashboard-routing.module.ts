import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { DashboardComponent } from './container/dashboard.component';
import { DataComponent } from './components/data/data.component';

const routes: Routes = [
  {
    path: '',
    component: DashboardComponent,
    children: [
      {
        path: 'data',
        component: DataComponent,
      },
    ],
  },
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule],
})
export class DashboardRoutingModule {}
