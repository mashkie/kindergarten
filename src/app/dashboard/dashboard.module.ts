import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

import { DashboardRoutingModule } from './dashboard-routing.module';
import { DashboardComponent } from './container/dashboard.component';
import { DataComponent } from './components/data/data.component';
import { ButtonComponent } from './components/button/button.component';

@NgModule({
  declarations: [DashboardComponent, DataComponent, ButtonComponent],
  imports: [CommonModule, DashboardRoutingModule],
})
export class DashboardModule {}
