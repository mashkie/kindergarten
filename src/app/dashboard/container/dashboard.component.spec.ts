import { ComponentFixture, TestBed } from '@angular/core/testing';

import { DashboardComponent } from './dashboard.component';
import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { DataComponent } from '../components/data/data.component';
import { ButtonComponent } from '../components/button/button.component';

describe('DashboardComponent', () => {
  let fixture: ComponentFixture<DashboardComponent>;
  let component: DashboardComponent;
  let spectator: Spectator<DashboardComponent>;

  const createComponent = createComponentFactory({
    component: DashboardComponent,
    declarations: [DataComponent, ButtonComponent],
    imports: [],
    providers: [],
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
    fixture = spectator.fixture;
    spectator.detectChanges();
  });

  it('should render the component', () => {
    expect(fixture).toMatchSnapshot();
  });
});
