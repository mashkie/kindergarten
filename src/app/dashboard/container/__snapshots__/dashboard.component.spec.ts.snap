// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent should render the component 1`] = `
<app-dashboard>
  <div
    class="dashboard container text-center"
  >
    <p>
      dashboard works!
    </p>
    <app-button>
      <button
        class="btn btn-primary"
        type="button"
      >
        Primary
      </button>
    </app-button>
    <app-data>
      <p>
        data works!
      </p>
    </app-data>
  </div>
</app-dashboard>
`;
